src/zarr/storage/common.py

```diff
@@ -46,6 +45,36 @@ def __init__(self, store: Store, path: str = "") -> None:
         self.store = store
         self.path = path
 
+    async def _init(self, mode: AccessModeLiteral) -> None:
+        """
+        Initialize the StorePath based on the provided mode.
+
+        * If the mode is 'w-' and the StorePath contains keys, raise a FileExistsError.
+        * If the mode is 'w', delete all keys nested within the StorePath
+        * If the mode is 'a', 'r', or 'r+', do nothing
+
+        Parameters
+        ----------
+        mode : AccessModeLiteral
+            The mode to use when initializing the store path.
+
+        Raises
+        ------
+        FileExistsError
+            If the mode is 'w-' and the store path already exists.
+        """
+        match mode:
```

Comment:
is this our first `match` statement in the codebase?

Reply:
perhaps! Do you like it?

Reply:
i do!